### PR TITLE
Add .h for c & c++

### PR DESCRIPTION
**Pull request by ievans**
Not sure if I need to run something after updating this

- [ ] I ran `make setup && make` to update the generated code after editing a `.atd` file (TODO: have a CI check)

### CODE REVIEW

**Review by emjin — APPROVED** [review]
Seems reasonable but I'd like to see the effect on our parse rate once this change is pulled in. It would be great if we could also have a test to make sure we parse .h files using c!

File: lang.json [hunk 1]
```diff
@@ -104,7 +105,8 @@
     "maturity": "alpha",
     "exts": [
       ".cc",
-      ".cpp"
+      ".cpp",
+      ".h"
```
Comment (jbergler) [inline lang.json:109]: Shouldn't this be `.hh` ?

Comment (aryx) [reply lang.json:109]: many C++ projects also use .h for their header.
But yes that means we will parse .h files twice, once for C, and once for C++.
But yes we could also add .hh

**Comment by aryx** [comment]
@emjin lang.json is now autogenerated from generate.py, so we should edit generate.py instead of lang.json!
We can't have comments in json, but maybe @mjambon we can add a fake first entry with a big "DO_NOT_EDIT_THIS_FILE"

**Pull request merged**